src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -2017,29 +2004,33 @@ private CloudDomain findDomainByName(String name) {
-        IntFunction<ListSharedDomainsRequest> pageRequestSupplier = page -> ListSharedDomainsRequest.builder()
-                                                                                                    .page(page)
-                                                                                                    .build();
-        return PaginationUtils.requestClientV2Resources(page -> delegate.sharedDomains()
+    private Flux<DomainResource> getDomainResources() {
+        IntFunction<ListDomainsRequest> pageRequestSupplier = page -> ListDomainsRequest.builder()
+                                                                                        .page(page)
+                                                                                        .build();
+        return PaginationUtils.requestClientV3Resources(page -> delegate.domainsV3()
                                                                         .list(pageRequestSupplier.apply(page)));
     }
 
-    private Flux<? extends Resource<PrivateDomainEntity>> getPrivateDomainResources() {
-        IntFunction<ListPrivateDomainsRequest> pageRequestSupplier = page -> ListPrivateDomainsRequest.builder()
-                                                                                                      .page(page)
-                                                                                                      .build();
-        return PaginationUtils.requestClientV2Resources(page -> delegate.privateDomains()
-                                                                        .list(pageRequestSupplier.apply(page)));
+    private Flux<DomainResource> getSharedDomainResources() {
+        return getDomainResources().filter(domain -> domain.getRelationships()
+                                                           .getOrganization()
+                                                           .getData() != null);
     }
 
-    private Flux<? extends Resource<PrivateDomainEntity>> getPrivateDomainResourcesByOrganizationGuid(UUID organizationGuid) {
-        IntFunction<ListOrganizationPrivateDomainsRequest> pageRequestSupplier = page -> ListOrganizationPrivateDomainsRequest.builder()
-                                                                                                                              .organizationId(organizationGuid.toString())
-                                                                                                                              .page(page)
-                                                                                                                              .build();
-        return PaginationUtils.requestClientV2Resources(page -> delegate.organizations()
-                                                                        .listPrivateDomains(pageRequestSupplier.apply(page)));
+    private Flux<DomainResource> getPrivateDomainResources() {
```

Comment:
Why not using delegate.privateDomains ?

Reply:
private_domains is a v2 api endpoint. The v3 api distinguishes shared and private domains by whether they have an "owning organization"